### PR TITLE
Tlbattributes

### PR DESCRIPTION
**Pull request by yizhang82**
Not strictly required for net standard 2.0 but nice to have for tlbimp created dll to not have dangling references.
This is the same PR as https://github.com/dotnet/coreclr/pull/10975 but done in corefx repo. 
@jkotas @danmosemsft PTAL

### CODE REVIEW

File: src/System.Runtime.InteropServices/src/System/Runtime/InteropServices/Attributes.cs [hunk 1]
```diff
@@ -129,4 +140,135 @@ public SetWin32ContextInIDispatchAttribute()
...
+        }
+        public TypeLibVarFlags Value { get { return _val; } }
+    }
+
+    [AttributeUsage(AttributeTargets.Assembly, Inherited = false)]
+    [System.Runtime.InteropServices.ComVisible(true)]
+    public sealed class TypeLibVersionAttribute : Attribute
+    {
+        internal int _major;
+        internal int _minor;
```
Comment (stephentoub) [inline src/System.Runtime.InteropServices/src/System/Runtime/InteropServices/Attributes.cs]: Could all of these `internal`s be `private`s instead?  You could use get-only auto-props then as well.

File: src/System.Runtime.InteropServices/src/System/Runtime/InteropServices/Attributes.cs [hunk 2]
```diff
@@ -129,4 +140,135 @@ public SetWin32ContextInIDispatchAttribute()
...
+        FUsesGetLastError   = 0x0080,
+        FDefaultCollelem    = 0x0100,
+        FUiDefault          = 0x0200,
+        FNonBrowsable       = 0x0400,
+        FReplaceable        = 0x0800,
+        FImmediateBind      = 0x1000,
+    }
+
+    [Serializable]
+    [Flags()]
```
Comment (stephentoub) [inline src/System.Runtime.InteropServices/src/System/Runtime/InteropServices/Attributes.cs]: Nit: the `()` in these isn't necessary

File: src/System.Runtime.InteropServices/src/System/Runtime/InteropServices/Attributes.cs [hunk 3]
```diff
@@ -95,6 +95,17 @@ public enum IDispatchImplType
         SystemDefinedImpl = 0,
     }
 
+    [AttributeUsage(AttributeTargets.Assembly, Inherited = false)]
+    public sealed class ImportedFromTypeLibAttribute : Attribute
+    {
+        internal String _val;
```
Comment (stephentoub) [inline src/System.Runtime.InteropServices/src/System/Runtime/InteropServices/Attributes.cs]: Nit: these `String`s can be `string`s

**Comment by safern** [comment]
test Innerloop CentOS7.1 Release x64 Build and Test (#18431)

File: src/System.Runtime.InteropServices/src/System/Runtime/InteropServices/Attributes.cs (hunk 1 above)
Comment (yizhang82) [inline src/System.Runtime.InteropServices/src/System/Runtime/InteropServices/Attributes.cs]: Sure. Done

File: src/System.Runtime.InteropServices/src/System/Runtime/InteropServices/Attributes.cs (hunk 2 above)
Comment (yizhang82) [inline src/System.Runtime.InteropServices/src/System/Runtime/InteropServices/Attributes.cs]: Removed

File: src/System.Runtime.InteropServices/src/System/Runtime/InteropServices/Attributes.cs (hunk 3 above)
Comment (yizhang82) [inline src/System.Runtime.InteropServices/src/System/Runtime/InteropServices/Attributes.cs]: Done

**Pull request merged**